source/gameengine/Rasterizer/RAS_Polygon.cpp

```diff
@@ -120,3 +121,19 @@ RAS_DisplayArray *RAS_Polygon::GetDisplayArray()
 {
 	return m_darray;
 }
+
+const float *RAS_Polygon::GetNormal()
+{
+	float faceNormal[3];
+	float verts[4][3];
+
+	for (int v = 0; v < m_numvert; v++)
+	{
+		verts[v][0] = GetVertex(v)->getXYZ()[0];
+		verts[v][1] = GetVertex(v)->getXYZ()[1];
+		verts[v][2] = GetVertex(v)->getXYZ()[2];
+	}
+
+	normal_poly_v3(faceNormal, verts, m_numvert);
+	return faceNormal;
```

Comment:
ohhhh, here you return a pointer to an array which will be freed just after the return.
Use MT_Vector to make a safe copy to return.